Assets/BossRoom/Scripts/Shared/ProfileManager.cs

```diff
@@ -16,32 +18,29 @@ public static class ProfileManager
 
         static string GetProfile()
         {
-#if UNITY_EDITOR
-
-            //The code below makes it possible for the clone instance to log in as a different user profile in Authentication service.
-            //This allows us to test services integration locally by utilising Parrelsync.
-            if (ClonesManager.IsClone())
+            var arguments = Environment.GetCommandLineArgs();
+            for (int i = 0; i < arguments.Length; i++)
             {
-                Debug.Log("This is a clone project.");
-                var customArguments = ClonesManager.GetArgument().Split(',');
+                if (arguments[i] == AuthProfileCommandLineArg)
+                {
+                    var profileId = arguments[i + 1];
+                    return profileId;
+                }
+            }
 
-                //second argument is our custom ID, but if it's not set we would just use some default.
+#if UNITY_EDITOR
 
-                var hardcodedProfileID = customArguments.Length > 1 ? customArguments[1] : "defaultCloneID";
+            // When running in the Editor and not a ParrelSync clone, make a unique ID
+            // from the Application.dataPath. This will work for cloning projects
+            // manually, or with Virtual Projects. Since only a single instance of
+            // the Editor can be open for a specific dataPath, uniqueness is ensured.
+            var hashedBytes = new MD5CryptoServiceProvider()
```

Comment:
wait, does this work for builds as well? Say I have two players running at once, won't they use the same path?

Reply:
My guess is that you build will be a specific platform and fall in the first case and not UNITY_EDITOR

Reply:
And so running two players, without the -AuthProfile arg would result in the same issue correct? Wouldn't it be best if we could have some local file where we save whether there's already a build running? I guess this could be done in another PR, I don't want to add scope to your work

Reply:
Yep, same thing would happen, they would both try to initialize with string.Empty.

It really depends what you want the behavior to be. From just reading quickly I assumed you'd want something deterministic.

But if that's not the case, and you always want different builds to use a new ID unless specified by the command line parameter, you can simply return new Guids every time.

That would be a simple addition, but it really depends on what you folks want the behavior to be :D

Reply:
mmm I see, so we'd have new users every single time we run the game?
I can see this breaking if/when we integrate UGS services like Cloud Save.
And even if our own session management, we're basing ourselves on auth id, which means a game client crashing wouldn't be able to continue its progress when connecting back.
And thinking of it, managing this with local files wouldn't be trivial either. If a client crashes and we reopen a new one, how do we know which of the locally saved profiles it was using?
I guess we could heartbeat in use GUIDs and then when reconnecting just reuse the oldest one. again not trivial and I don't want to explode the scope of this PR

Reply:
In this case I would suggest merging this and addressing the scenario in the future in the following ways:

1. If you don't care about creating new users every time, you can generate random IDs.
2. If you want to reuse specific IDs, you need a way to manage them and assign them to game instances, whether they're Editor instances or player builds.